### PR TITLE
adding postgresql server facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding missing PostgreSQL server facts.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_postgresqlserver_facts

##### ANSIBLE VERSION
2.6

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+short_description: Get PostgreSQL Server facts.
+description:
+    - Get facts of PostgreSQL Server.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: Let's just standardize on `resource_group` and `name` for these (feel free to leave `server_name` as an alias though); the resource type should already be clear from the module name in a playbook context

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                - Server version.
+            returned: always
+            type: str
+            sample: version
+        user_visible_state:
+            description:
+                - A state of a server that is visible to user.
+            returned: always
+            type: str
+            sample: user_visible_state
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: not sure what this is?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        fully_qualified_domain_name:
+            description:
+                - The fully qualified domain name of a server.
+            returned: always
+            type: str
+            sample: fully_qualified_domain_name
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: real sample value

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+                        - The scale capacity."
+                    returned: always
+                    type: int
+                    sample: 100
+        version:
+            description:
+                - Server version.
+            returned: always
+            type: str
+            sample: version
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: real sample value

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+                    sample: PGSQLB100
+                tier:
+                    description:
+                        - The tier of the particular SKU
+                    returned: always
+                    type: str
+                    sample: Basic
+                capacity:
+                    description:
+                        - The scale capacity."
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: spurious quote?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                - Resource name.
+            returned: always
+            type: str
+            sample: testserver
+        location:
+            description:
+                - The location the resource resides in.
+            returned: always
+            type: str
+            sample: onebox
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: real sample location?

**Review by nitzmahone — CHANGES_REQUESTED** [review]
A couple minor docs tweaks and this one's good to go

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/77921/75/tests

**Comment by Fred-sun** [comment]
@zikalino has fixed test according with comment. @mattclay @nitzmahone Could you help review again？ Thanks!

**Comment by Fred-sun** [comment]
@mattclay @nitzmahone Thanks for your contribution, Could you help review again？

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_postgresqlserver_facts
+version_added: "2.7"
+short_description: Get PostgreSQL Server facts.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: azure

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        self.name = None
+        super(AzureRMServersFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(PostgreSQLManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if (self.resource_group is not None and
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: how to get per tags?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+        )
+        self.mgmt_client = None
+        self.resource_group = None
+        self.name = None
+        super(AzureRMServersFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(PostgreSQLManagementClient,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: pls, put all managemt  client into azure_rm_common

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+                type='str'
+            ),
+            tags=dict(
+                type='list'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict()
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlserver_facts.py]: ansible_facts?

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Comment by zikalino** [comment]
shipit

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR? Thanks!

**Comment by Fred-sun** [comment]
@nitzmahone Could you help review this PR? Thanks!

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**